### DISCUSSION TITLE
how to align an element to the center?

### DISCUSSION THREAD

**Discussion by EdsonMateus1** [fabricjs/fabric.js#7771] [Q&A] [answered]
how to align an element to the center?

**Comment by eriqjaffe** [comment]
What I've done...

```
object = new fabric.Rect({
	originX: 'center',
	originY: 'center',
	fill: 'red',
	height: 100,
	width: 100
})

canvas.add(object)
canvas.centerObject(object)
```

To be honest I'm not sure if the originX and originY are needed just to center something.

**Comment by ShaMan123** [comment, marked as answer]
centering an object on canvas
```js
obj.center()
obj.centerH()
obj.centerV()

```

centering an object according to viewport
```obj.viewportCenter()``` etc.
